include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,267 @@
+//======================================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+#include <eve/logical.hpp>
+#include <eve/concept/options.hpp>
+#include <eve/conditional.hpp>
+
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct options
+  //!   @brief  Wrapper class around bundle of options passed to eve::callable
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   eve::callable can be decorated with options. eve::options is used internally to gather all those options
+  //!   in a @raberu settings instance for further processing.
+  //!
+  //!   @tparam Settings @raberu settings type to wrap.
+  //! @}
+  //====================================================================================================================
+  template <rbr::concepts::settings Settings = rbr::settings<>>
+  struct options : Settings
+  {
+    constexpr EVE_FORCEINLINE options() : Settings{} {}
+
+    template <rbr::concepts::option... Options>
+    constexpr EVE_FORCEINLINE explicit options(Options && ... opts) : Settings(EVE_FWD(opts) ...) {}
+
+    template <typename... Options>
+    constexpr EVE_FORCEINLINE explicit options(rbr::settings<Options...> const& opts) : Settings(opts) {}
+  };
+
+  template <rbr::concepts::option ... Options>
+  options(Options&& ... opts) -> options<decltype(rbr::settings(EVE_FWD(opts) ...))>;
+
+  template <typename... Options>
+  options(rbr::settings<Options...> const&)  -> options<rbr::settings<Options...>>;
+}
+
+//======================================================================================================================
+// Overload's helpers
+//======================================================================================================================
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @var functor
+  //!   @brief EVE's @callable generator
+  //! @}
+  //====================================================================================================================
+  template<template<typename> class Func>
+  inline constexpr auto functor = Func<eve::options<>>{};
```

Comment:
What is this guy?

Reply:
The thing we use to define actual functor with a proper default options.
We verbally agreed on having this last time.